update to 0.1.0 - Add new serialization settings and unit tests

nmorgan-cb opened the pull request.

nmorgan-cb committed:
update to 0.1.0 - Add new serialization settings and unit tests

nmorgan-cb committed:
combine redundant classes

nmorgan-cb committed:
Update license headers

cfluke-cb commented on src/CoinbaseSdk/Core/credentials/CoinbaseCredentials.cs:
```diff
@@ -32,19 +32,20 @@ public class CoinbaseCredentials
     /// <param name="accessKey">Coinbase API AccessKey.</param>
     /// <param name="passphrase">Coinbase API Secret Key.</param>
     /// <param name="signingKey">Coinbase API Signing Key.</param>
+    [System.Diagnostics.CodeAnalysis.SetsRequiredMembers]
```
I don't think this is needed

cfluke-cb commented on GEMINI.md:
```diff
@@ -0,0 +1,53 @@
+# Coinbase .NET Core
```
does this need to be committed?

cfluke-cb commented on AGENTS.md:
```diff
@@ -0,0 +1,19 @@
+# Repository Guidelines
```
does this need to be committed?

cfluke-cb commented on CHANGELOG.md:
```diff
@@ -0,0 +1,20 @@
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [0.1.0] - 2025-11-13
```
I would forward date this, as it is already off

nmorgan-cb committed:
remove agents and gemini files from source control

nmorgan-cb committed:
Update to use raw strings and @ string syntax over escaped strings

nmorgan-cb committed:
Update to consistent formatting for json strings

nmorgan-cb committed:
fix required params and run .NET format

nmorgan-cb committed:
update format of json strings

nmorgan-cb committed:
Add error test cases

nmorgan-cb committed:
bump changelog date

cfluke-cb commented on src/CoinbaseSdk/Core/credentials/CoinbaseCredentials.cs:
```diff
@@ -16,86 +16,46 @@
-      string accessKey = null,
-      string passphrase = null,
-      string signingKey = null)
+    public class CoinbaseCredentials(string accessKey, string passphrase, string signingKey)
```
this will not error if they pass in null as any of these values. you want to only allow the public to be get, and use an exception in the constructor to set

public class CoinbaseCredentials
{
    public string AccessKey { get; }
    public string Passphrase { get; }
    public string SigningKey { get; }

    public CoinbaseCredentials(string accessKey, string passphrase, string signingKey)
    {
        AccessKey = accessKey ?? throw new ArgumentNullException(nameof(accessKey));
        Passphrase = passphrase ?? throw new ArgumentNullException(nameof(passphrase));
        SigningKey = signingKey ?? throw new ArgumentNullException(nameof(signingKey));
    }
}

nmorgan-cb committed:
Ensure ArgumentNullException thrown when credentials parameters aren't passed

nmorgan-cb committed:
fix serialization bugs

nmorgan-cb committed:
fix defaults

rcbgr commented on src/CoinbaseSdk/Core/CoinbaseSdk.Core.csproj:
```diff
@@ -2,8 +2,8 @@
-    <LangVersion>11</LangVersion>
+    <Version>0.1.0</Version>
+    <LangVersion>12</LangVersion>
     <Authors>Coinbase Solutions Architecture</Authors>
```
Change this to Coinbase

nmorgan-cb commented on src/CoinbaseSdk/Core/http/SystemNetHttpClient.cs:
```diff
@@ -16,194 +16,194 @@
+                return false;
+            }
+
+            if (requestException)
```
I think these got nailed by a dotnet format as this hasn't been changed in the PR from a business logic standpoint

I don't think we can guarantee that all write operations are non-retryable. There are a few data fetching endpoints that are POST requests. We also have no way of knowing if the request contains an idempotency key as that would need all of our packages to have consistent approaches to idempotency.

I'll add the ability to turn off retries so that it can be configured for certain key endpoints, but I don't think a blanket "writes can't retry" setup is effective.

nmorgan-cb committed:
Refactor Core SDK for v0.1.0: extensibility, thread safety, and simplified retries

Architecture & Refactoring:
- Introduce protected virtual hooks (`BuildRequest`, `ConfigureRequest`, `SendHttpRequestAsync`, `ValidateResponse`) in `CoinbaseClient` to enable cleaner subclass customizations.
- Simplify `CoinbasePrimeClient` by leveraging these new base class hooks, significantly reducing code duplication.
- Remove `IRetryPolicyProvider` abstraction; `SystemNetHttpClient` now integrates directly with Polly retry policies.

Configuration & Reliability:
- Add `HttpClientDefaults` and `RetryPolicyDefaults` for centralized, consistent default values.
- Update `JsonUtility` to use `Lazy<T>` for robust, thread-safe static initialization.
- Fix potential `NullReferenceException` in `CoinbaseCredentials` constructor.

Testing & Documentation:
- Add `StubHttpMessageHandler` and comprehensive unit tests for client logic, credentials, and serialization.
- Update `CHANGELOG.md` and `README.md` to reflect v0.1.0 features, configuration guides, and architectural improvements.

nmorgan-cb committed:
update author

nmorgan-cb committed:
update license headers

nmorgan-cb committed:
remove blank line

nmorgan-cb committed:
update core with safety checks
